src/custom/Modal/index.tsx

```diff
@@ -172,7 +173,11 @@ export const ModalFooter: React.FC<ModalFooterProps> = ({ helpText, children, va
 // ModalButtonPrimary
 export const ModalButtonPrimary: React.FC = styled(ContainedButton)(({ theme }) => ({
   backgroundColor: theme.palette.background.brand?.default,
-  color: theme.palette.text.constant?.white
+  color: theme.palette.text.constant?.white,
+  '&.MuiButton-contained.Mui-disabled': {
+    color: charcoal[50],
```

Comment:
nope, it should be like above , see line 176- theme.palette....
Otherwise the dark and light mode changes will not take place